test/request_handler_proxy_test.cpp

```diff
@@ -0,0 +1,50 @@
+#include <string>
+#include "gtest/gtest.h"
+#include "request_handler_proxy.hpp"
+#include "request_handler.hpp"
+#include "request.hpp"
+#include "request_parser.hpp"
+#include "http_client.hpp"
+#include "response.hpp"
+#include "../nginx-configparser/config_parser.h"
+
+class ProxyHandlerTest : public ::testing::Test {
+protected:
+	bool parseString(const std::string config_string) {
+		std::stringstream config_stream(config_string);
+		return parser.Parse(&config_stream, &out_config);
+	}
+
+	http::server::RequestHandler::Status initProxy(std::string prefix){
+		return proxy_handler.Init(prefix, out_config);
+	}
+
+    void clearConfig() {
+        out_config = NginxConfig();
+    }
+
+	NginxConfigParser parser;
+	NginxConfig out_config;
+	http::server::ProxyHandler proxy_handler;
+};
+
+//test init of ProxyHandler with various configs
+TEST_F(ProxyHandlerTest, InitTest) {
```

Comment:
yeah.. it is since it mostly consists of a bunch of private member function calls, and on top of that alot of boost library calls within thouse functions. I left a comment at the bottom of this file to make note of this